samtranslator/swagger/swagger.py

```diff
@@ -781,6 +782,41 @@ def _set_method_apikey_handling(self, path, method_name, apikey_required):
             if security != existing_security:
                 method_definition["security"] = security
 
+    def add_request_validator_to_method(self, path, method_name, validate_body=False, validate_request=False):
+        """
+        Adds request model body parameter for this path/method.
+
+        :param string path: Path name
+        :param string method_name: Method name
+        :param bool validate_body: Add validator parameter on the body
+        :param bool validate_request: Validate request
+        """
+
+        normalized_method_name = self._normalize_method_name(method_name)
+        validator_name = SwaggerEditor.get_validator_name(validate_body, validate_request)
+
+        # Creating validator
+        request_validator_definition = {
+            validator_name: {"validateRequestBody": validate_body, "validateRequestParameters": validate_request}
+        }
+        if not self._doc.get(self._X_APIGW_REQUEST_VALIDATOR):
+            self._doc[self._X_APIGW_REQUEST_VALIDATOR] = {}
+
+        if not self._doc[self._X_APIGW_REQUEST_VALIDATOR].get(validator_name):
+            # Adding only if the validator hasn't been defined already
+            self._doc[self._X_APIGW_REQUEST_VALIDATOR].update(request_validator_definition)
+
+        # It is possible that the method could have two definitions in a Fn::If block.
+        for method_definition in self.get_method_contents(self.get_path(path)[normalized_method_name]):
```

Comment:
Question here: would the swagger doc ever cause self.get_path(path)[normalized_method_name] to error out? Any checks needed?

Reply:
Hi @qingchm tbh if that happen w9uld happen way before than set the validators. As the models and other methods setting apigw methods is doing it and that validations would happen before. I am not on laptop right now but I can chek on the code where it is being checked.

Reply:
Hi @qingchm Sorry getting back to you only now. So, looking at this, the only way of this fails is if the Method is not defined, and it will fail waay before as I said, once per [this line](https://github.com/aws/serverless-application-model/blob/90edf3e1304f79bb5da12860e6fe6e6c776ff5a5/samtranslator/model/stepfunctions/events.py#L236) shows the model as required. The normalized method get is expect only a small conversion and the [get_path method](https://github.com/aws/serverless-application-model/blob/90edf3e1304f79bb5da12860e6fe6e6c776ff5a5/samtranslator/swagger/swagger.py#L52) would return its path dict (where the Method is required). :)